File: frontend/lib/models/photo.dart
class Photo {
  final int id;
  final int fuiteId;
  final String cheminFichier;
  final String? thumbnailUrl;
  final String? datePrise;
  final String? annotationsDessin;

  Photo({
    required this.id,
    required this.fuiteId,
    required this.cheminFichier,
    this.thumbnailUrl,
    this.datePrise,
    this.annotationsDessin,
  });

  factory Photo.fromJson(Map<String, dynamic> json) {
    return Photo(
      id: json['id'] as int,
      fuiteId: json['fuiteId'] as int,
      cheminFichier: json['cheminFichier'] as String,
      thumbnailUrl: json['thumbnailUrl'] as String?,
      datePrise: json['datePrise'] as String?,
      annotationsDessin: json['annotationsDessin'] as String?,
    );
  }
}

File: backend/src/main/java/com/backend/backend/dao/entities/Photo.java
package com.backend.backend.dao.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.util.Date;

@Entity
@Table(name = "photos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Photo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String cheminFichier;

    private String thumbnailUrl;

    @Temporal(TemporalType.DATE)
    private Date datePrise;

    // JSON des tracés manuels (annotations dessin)
    @Lob
    @Column(columnDefinition = "TEXT")
    private String annotationsDessin;

    @ManyToOne
    @JoinColumn(name = "fuite_id", nullable = false)
    private Fuite fuite;
}
